shared/oae/api/oae.api.util.js

```diff
@@ -960,10 +968,24 @@ define(['exports', 'require', 'jquery', 'underscore', 'oae.api.config', 'markdow
                     // Get the query from the request URL on the Ajax object, as that is the only provided clue
                     // for finding out the search query
                     var query = $.url(this.url).param('q');
+                    
+                    // Track any results that need to be excluded
+                    var toExclude = [];
+
                     $.each(data.results, function(index, result) {
-                        result.displayName = security().encodeForHTML(result.displayName);
-                        result.query = query;
+                        if (_.indexOf(options.exclude, result.id) !== -1) {
+                            toExclude.push(index);
+                        } else {
+                            result.displayName = security().encodeForHTML(result.displayName);
+                            result.query = query;
+                        }
                     });
+                    
+                    // Remove excluded items
+                    while (toExclude.length > 0) {
+                        data.results.splice(_.last(toExclude), 1);
+                        toExclude.pop();
```

Comment:
We can get rid of a line if you just do `data.results.splice(toExclude.pop(), 1);`. But understand if you don't want to go that route since it may conceal the mutation of the array.